src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -70,45 +71,74 @@ class F extends ValidationTest {
 
 export const g = makeTestGroup(F);
 
-g.test('validation_of_sampleCount')
-  .params([
-    // TODO: Consider making a list of "valid"+"invalid" texture descriptors in capability_info.
-    { sampleCount: 0, _success: false }, // sampleCount of 0 is not allowed
-    { sampleCount: 1, _success: true }, // sampleCount of 1 is allowed
-    { sampleCount: 2, _success: false }, // sampleCount of 2 is not allowed
-    { sampleCount: 3, _success: false }, // sampleCount of 3 is not allowed
-    { sampleCount: 4, _success: true }, // sampleCount of 4 is allowed
-    { sampleCount: 8, _success: false }, // sampleCount of 8 is not allowed
-    { sampleCount: 16, _success: false }, // sampleCount of 16 is not allowed
-    { sampleCount: 4, mipLevelCount: 2, _success: false }, // multisampled multi-level not allowed
-    { sampleCount: 4, arrayLayerCount: 2, _success: false }, // multisampled multi-layer is not allowed
-  ])
+g.test('mipLevelCount_validation_on_formats')
+  .params(poptions('format', kAllTextureFormats))
   .fn(async t => {
-    const { sampleCount, mipLevelCount, arrayLayerCount, _success } = t.params;
+    const format: GPUTextureFormat = t.params.format;
 
-    const descriptor = t.getDescriptor({ sampleCount, mipLevelCount, arrayLayerCount });
+    const descriptor = {
+      size: { width: 32, height: 32, depth: 1 },
+      mipLevelCount: 6,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
 
-    t.expectValidationError(() => {
-      t.device.createTexture(descriptor);
-    }, !_success);
+    // const extension: GPUExtensionName | undefined = kAllTextureFormatInfo[format].extension;
+    const extension: GPUExtensionName = kAllTextureFormatInfo[format].extension;
```

Comment:
I keep get error info like below:
Type '"texture-compression-bc" | undefined' is not assignable to type 'GPUExtensionName'

I tried a few methods (like the code on line 86), but they don't work. 

Any graceful solution for this if I don't split the tests into two: one for Uncompressed texture format (no extension needed), another for Compressed texture format (need extension)?  

In addition, it is valid to create mipmaps for compressed texture? I simply looked into Vulkan, looks like it is OK because it doesn't say no.

Reply:
Thanks for your update!